base/regex.jl

```diff
@@ -364,12 +364,25 @@ function endswith(s::SubString{String}, r::Regex)
 end
 
 function chopprefix(s::AbstractString, prefix::Regex)
+    pattern = prefix.pattern
+    
+    # fast path for letter-only regexes
+    all(isletter, pattern) && startswith(s, pattern) && return chopprefix(s, pattern)
```

Comment:
I don't like the look of this "fast path".  `isletter` requires a Unicode category lookup for every character, and doing `startswith` followed by `chopprefix` runs a `startswith` check twice.

If we want to optimize case-insensitive non-regex matches, which should add `ignorecase` options to `startswith` and `chopprefix` etcetera.